activerecord/lib/active_record/database_configurations.rb

```diff
@@ -73,13 +73,11 @@ def find_db_config(env)
       end
     end
 
-    # Returns the DatabaseConfig object as a Hash.
+    # Returns the DatabaseConfigurations object as a Hash.
     def to_h
-      configs = configurations.reverse.inject({}) do |memo, db_config|
+      configurations.inject({}) do |memo, db_config|
         memo.merge(db_config.to_legacy_hash)
       end
-
-      Hash[configs.to_a.reverse]
```

Comment:
Can you put both of these back? They are not there accidentally. I'll add a test showing why they both need to be there but the gist is if the first reverse isn't there a multi-db tier will return the last configs instead of the first configs (ie animals instead of primary) and the second one was just for parity with Rails 5.2 - that it will return the config hash in the order of the database.yml. This is for backwards compatibility and will be removed in 6.1.